src/pkg/egress/syslog/https_batch.go

```diff
@@ -0,0 +1,117 @@
+package syslog
+
+import (
+	"bytes"
+	"crypto/tls"
+	"time"
+
+	"code.cloudfoundry.org/go-loggregator/v9/rpc/loggregator_v2"
+	metrics "code.cloudfoundry.org/go-metric-registry"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress"
+)
+
+const BATCHSIZE = 256 * 1024
+
+type HTTPSBatchWriter struct {
+	HTTPSWriter
+	msgBatch     bytes.Buffer
+	batchSize    int
+	sendInterval time.Duration
+	sendTimer    *TriggerTimer
+	egrMsgCount  float64
+}
+
+func NewHTTPSBatchWriter(
+	binding *URLBinding,
+	netConf NetworkTimeoutConfig,
+	tlsConf *tls.Config,
+	egressMetric metrics.Counter,
+	c *Converter,
+) egress.WriteCloser {
+	client := httpClient(netConf, tlsConf)
+	binding.URL.Scheme = "https" // reset the scheme for usage to a valid http scheme
+	return &HTTPSBatchWriter{
+		HTTPSWriter: HTTPSWriter{
+			url:             binding.URL,
+			appID:           binding.AppID,
+			hostname:        binding.Hostname,
+			client:          client,
+			egressMetric:    egressMetric,
+			syslogConverter: c,
+		},
+		batchSize:    BATCHSIZE,
+		sendInterval: 1 * time.Second,
+		egrMsgCount:  0,
+	}
+}
+
+func (w *HTTPSBatchWriter) sendMsgBatch() error {
+	currentEgrCount := w.egrMsgCount
+	currentMsg := w.msgBatch.Bytes()
+
+	w.egrMsgCount = 0
+	w.msgBatch.Reset()
+
+	return w.sendHttpRequest(currentMsg, currentEgrCount)
+}
+
+// Modified Write function
+func (w *HTTPSBatchWriter) Write(env *loggregator_v2.Envelope) error {
+	msgs, err := w.syslogConverter.ToRFC5424(env, w.hostname)
+	if err != nil {
+		return err
+	}
+
+	for _, msg := range msgs {
+		w.msgBatch.Write(msg)
+		w.egrMsgCount += 1
+		w.startAndTriggerSend()
+	}
+	return nil
+}
+
+// TODO: Error back propagation. Errors are not looked at further down the call chain
+func (w *HTTPSBatchWriter) startAndTriggerSend() {
+	if w.sendTimer == nil || !w.sendTimer.Running() {
```

Comment:
As an alternative to this approach you could do something similar to the ingress client example and have a channel to communicate with a sending goroutine that does a select on that channel and a timer channel.

Reply:
As already explained here:
https://github.com/cloudfoundry/loggregator-agent-release/pull/491#discussion_r1646978226
I am as told agnostic to the final implementation, I just found the encapsulating into a class handling the logic more understandable.

Reply:
reimplemented with channel and timer channel